### PR TITLE
[INLONG-8676][Manager] Elasticsearch - Modify the calling method from SDK to HTTP 

### LINKED ISSUE #8676: [Improve][Manager] Elasticsearch - Modify the calling method from SDK to HTTP
### Description

The available API could refer to https://www.elastic.co/guide/en/elasticsearch/reference/current/rest-apis.html.

### InLong Component

InLong Manager

### Are you willing to submit PR?

- [ ] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by haibo-duan**
### Prepare a Pull Request
*(Change the title refer to the following example)*

- Title Example: [INLONG-XYZ][Component] Title of the pull request

*(The following *XYZ* should be replaced by the actual [GitHub Issue](https://github.com/apache/inlong/issues) number)*

- Fixes #8676 

### Motivation

*Explain here the context, and why you're making that change. What is the problem you're trying to solve?*

### Modifications

Modify the elasticsearch calling method from SDK to HTTP 

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AuditServiceImpl.java [hunk 1]
```diff
@@ -365,27 +362,36 @@ private List<String> getAuditIds(String groupId, String streamId, String sourceN
...
+        query.add("bool", bool);
+        result.add("query", query);
+        final String sortStr = "[{\"log_ts\": {\"order\": \"asc\"}}]";
+        result.add("sort", GSON.fromJson(sortStr, JsonArray.class));
+        final String aggregationsStr = "{\"log_ts\": {\"terms\": {\"field\": \"log_ts\",\"size\": 2147483647, "
+                + "\"min_doc_count\": 1,\"shard_min_doc_count\": 0, \"show_term_doc_count_error\": false,"
+                + "\"order\":[{\"_count\": \"desc\" },{\"_key\": \"asc\" }]},\"aggregations\": "
+                + "{\"count\":{\"sum\":{\"field\":\"count\" }},\"delay\": {\"sum\":{\"field\":\"delay\"}}}}}";
+        result.add("aggregations", GSON.fromJson(aggregationsStr, JsonElement.class));
+        return result;
```
Comment (vernedeng) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AuditServiceImpl.java:394]: Why so many magic numbers

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/util/HttpUtils.java [hunk 2]
```diff
@@ -95,6 +95,25 @@ public static <T> T request(RestTemplate restTemplate, String url, HttpMethod me
...
+        boolean result = false;
+        HttpEntity<String> request = new HttpEntity(params, header);
+        log.debug("send request to {}, param {}", url, params);
+        exchange = restTemplate.exchange(url, HttpMethod.HEAD, request, String.class);
+        HttpStatus statusCode = exchange.getStatusCode();
+        if (statusCode.is2xxSuccessful()) {
+            result = true;
+        } else {
+            log.debug("success request to {},  status code {}, body {}", url, statusCode, exchange.getBody());
+        }
```
Comment (fuweng11) [inline inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/util/HttpUtils.java:113]:
```suggestion
        if (statusCode.is2xxSuccessful()) {
            result = true;
        }
        log.debug("success request to {},  status code {}, body {}", url, statusCode, exchange.getBody());
        
```

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/es/ElasticsearchConfig.java [hunk 3]
```diff
@@ -45,7 +39,8 @@
 public class ElasticsearchConfig {
 
     private static final Logger logger = LoggerFactory.getLogger(ElasticsearchConfig.class);
-    private static RestHighLevelClient highLevelClient;
+    private static RestTemplate restTemplate;
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/es/ElasticsearchConfig.java:42]: Why not use 
`@Autowired
   private RestTemplate restTemplate;` ？

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/util/HttpUtils.java [hunk 4]
```diff
@@ -95,6 +95,24 @@ public static <T> T request(RestTemplate restTemplate, String url, HttpMethod me
...
+    public static boolean headRequest(RestTemplate restTemplate, String url, Map<String, Object> params,
+            HttpHeaders header) {
+        ResponseEntity<String> exchange;
+        boolean result = false;
+        HttpEntity<String> request = new HttpEntity(params, header);
+        log.debug("send request to {}, param {}", url, params);
+        exchange = restTemplate.exchange(url, HttpMethod.HEAD, request, String.class);
+        HttpStatus statusCode = exchange.getStatusCode();
+        if (statusCode.is2xxSuccessful()) {
+            result = true;
```
Comment (vernedeng) [inline inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/util/HttpUtils.java:110]:
```suggestion
            result = statusCode.is2xxSuccessful()
```

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AuditServiceImpl.java [hunk 5]
```diff
@@ -295,18 +298,21 @@ public List<AuditVO> listByCondition(AuditRequest request) throws Exception {
...
                     continue;
                 }
-                SearchResponse response = elasticsearchApi.search(toAuditSearchRequest(index, groupId, streamId));
-                final List<Aggregation> aggregations = response.getAggregations().asList();
-                if (CollectionUtils.isNotEmpty(aggregations)) {
-                    ParsedTerms terms = (ParsedTerms) aggregations.get(0);
-                    if (CollectionUtils.isNotEmpty(terms.getBuckets())) {
-                        List<AuditInfo> auditSet = terms.getBuckets().stream().map(bucket -> {
+                JsonObject response = elasticsearchApi.search(index, toAuditSearchRequestJson(groupId, streamId));
+                JsonObject aggregations = response.getAsJsonObject("aggregations").getAsJsonObject("log_ts");
```
Comment (vernedeng) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AuditServiceImpl.java:302]: make these field name as constants

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/es/ElasticsearchApi.java [hunk 6]
```diff
@@ -17,98 +17,131 @@
...
-        getIndexRequest.indices(indexName);
-        return getEsClient().indices().exists(getIndexRequest, RequestOptions.DEFAULT);
+    public boolean indexExists(String indexName) throws Exception {
+        final String url = esConfig.getOneHttpUrl() + "/" + indexName;
+        try {
+            return HttpUtils.headRequest(esConfig.getRestClient(), url, null, getHttpHeaders());
+        } catch (NotFound e) {
+            return false;
+        } catch (Exception e) {
+            throw e;
```
Comment (vernedeng) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/es/ElasticsearchApi.java:115]: do something,  or not catch this expcetion

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AuditServiceImpl.java [hunk 7]
```diff
@@ -295,18 +302,23 @@ public List<AuditVO> listByCondition(AuditRequest request) throws Exception {
                     LOGGER.warn("elasticsearch index={} not exists", index);
                     continue;
                 }
-                SearchResponse response = elasticsearchApi.search(toAuditSearchRequest(index, groupId, streamId));
-                final List<Aggregation> aggregations = response.getAggregations().asList();
-                if (CollectionUtils.isNotEmpty(aggregations)) {
-                    ParsedTerms terms = (ParsedTerms) aggregations.get(0);
-                    if (CollectionUtils.isNotEmpty(terms.getBuckets())) {
-                        List<AuditInfo> auditSet = terms.getBuckets().stream().map(bucket -> {
+                JsonObject response = elasticsearchApi.search(index, toAuditSearchRequestJson(groupId, streamId));
```
Comment (vernedeng) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AuditServiceImpl.java:305]: Is there any more elegent way to process json from es Api?

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AuditServiceImpl.java [hunk 8]
```diff
@@ -365,27 +377,43 @@ private List<String> getAuditIds(String groupId, String streamId, String sourceN
...
-        filterBuilder.must(termQuery("inlong_group_id", groupId));
-        filterBuilder.must(termQuery("inlong_stream_id", streamId));
-        SearchSourceBuilder sourceBuilder = new SearchSourceBuilder();
-        sourceBuilder.aggregation(builder);
-        sourceBuilder.query(filterBuilder);
-        sourceBuilder.from(0);
-        sourceBuilder.size(0);
-        sourceBuilder.sort("log_ts", SortOrder.ASC);
-        return new SearchRequest(new String[]{index}, sourceBuilder);
+    public static JsonObject toAuditSearchRequestJson(String groupId, String streamId) {
```
Comment (vernedeng) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AuditServiceImpl.java:386]: ditto

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/es/ElasticsearchApi.java [hunk 9]
```diff
@@ -147,93 +182,101 @@ private List<String> getMappingInfo(List<ElasticsearchFieldInfo> fieldsInfo) {
...
         List<String> fieldList = getMappingInfo(fieldInfos);
-        StringBuilder mapping = new StringBuilder().append("{\n      \"properties\" : {\n")
-                .append(StringUtils.join(fieldList, ",\n")).append("\n      }\n}");
-        createIndexRequest.mapping(mapping.toString(), XContentType.JSON);
-
-        CreateIndexResponse createIndexResponse = getEsClient().indices()
-                .create(createIndexRequest, RequestOptions.DEFAULT);
-        LOG.info("create {}:{}", indexName, createIndexResponse.isAcknowledged());
+        StringBuilder mapping = new StringBuilder().append("\n{\"mappings\":\n{\n\"properties\":{\n")
+                .append(StringUtils.join(fieldList, ",\n")).append("\n}\n}\n}");
```
Comment (vernedeng) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/es/ElasticsearchApi.java:194]: ditto

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/es/ElasticsearchConfig.java [hunk 10]
```diff
@@ -45,7 +40,11 @@
 public class ElasticsearchConfig {
 
     private static final Logger logger = LoggerFactory.getLogger(ElasticsearchConfig.class);
-    private static RestHighLevelClient highLevelClient;
+
+    @Autowired
+    private RestTemplate restTemplate;
+
+    private static List<HttpHost> httpHosts;
```
Comment (vernedeng) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/es/ElasticsearchConfig.java:47]: why use static variables, is there any concurrency issue?

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AuditServiceImpl.java [hunk 11]
```diff
@@ -104,13 +106,32 @@
...
+    private static final int ES_QUERY_SIZE = 0;
+    private static final String ES_SORT_ORDER = "ASC";
+    private static final String ES_TERM_FILED = "log_ts";
+    private static final String ES_AGGREGATIONS_COUNT = "count";
+    private static final String ES_AGGREGATIONS_DELAY = "delay";
+    private static final String ES_AGGREGATIONS = "aggregations";
+    private static final String ES_BUCKETS = "buckets";
+    private static final String ES_KEY = "key";
+    private static final String ES_VALUE = "value";
+    private static final String ES_INLONG_GROUP_ID = "inlong_group_id";
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AuditServiceImpl.java:129]: No need to add ES prefix, these parameters are independent of ES.

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/es/ElasticsearchConfig.java [hunk 12]
```diff
@@ -56,54 +58,84 @@ public class ElasticsearchConfig {
...
+     * @return a http url
+     * @throws Exception any exception if occurred
+     */
+    public String getOneHttpUrl() throws Exception {
+        getHttpHosts();
+        if (!httpHosts.isEmpty() && httpHosts.size() > 0) {
+            return httpHosts.get(rand.nextInt(httpHosts.size())).toString();
+        } else {
+            throw new Exception("http hosts is empty! please check hosts!");
         }
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/es/ElasticsearchConfig.java:119]:
```suggestion
        getHttpHosts();
        if (!httpHosts.isEmpty() && httpHosts.size() > 0) {
            return httpHosts.get(rand.nextInt(httpHosts.size())).toString();
        }
        throw new Exception("http hosts is empty! please check hosts!");
        
```

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/es/ElasticsearchConfig.java [hunk 13]
```diff
@@ -56,54 +58,84 @@ public class ElasticsearchConfig {
...
+        getHttpHosts();
+        if (!httpHosts.isEmpty() && httpHosts.size() > 0) {
+            String[] urls = new String[httpHosts.size()];
+            for (int i = 0; i < urls.length; i++) {
+                urls[i] = httpHosts.get(i) + urlSuffix;
+            }
+            return urls;
+        } else {
+            throw new Exception("http hosts is empty! please check hosts!");
+        }
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/es/ElasticsearchConfig.java:139]:
```suggestion
        getHttpHosts();
        if (!httpHosts.isEmpty() && httpHosts.size() > 0) {
            String[] urls = new String[httpHosts.size()];
            for (int i = 0; i < urls.length; i++) {
                urls[i] = httpHosts.get(i) + urlSuffix;
            }
            return urls;
        }
        throw new Exception("http hosts is empty! please check hosts!");
        
```

**Review by fuweng11 — APPROVED** [review]

**Pull request merged**